### PR TITLE
RELATED: BB-1490 fixed sticky row error when table is hidden

### PR DESCRIPTION
**Pull request by yanlipnican**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-app-components:
- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by yanlipnican** [comment]
extended test - examples

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3284/ : SUCCESS in 8m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4066/ : SUCCESS in 12s

**Comment by yanlipnican** [comment]
extended test - storybook

File: src/components/core/tests/PivotTable.spec.tsx [hunk 1]
```diff
@@ -294,6 +295,45 @@ describe("PivotTable", () => {
...
+            tableInstance.setState({ columnDefs: [{}] });
+            expect(tableInstance.isTableHidden()).toEqual(false);
+        });
+    });
+
+    describe("onModelUpdated", () => {
+        const wrapper = renderComponent();
+        const table = wrapper.find(PivotTableInner);
+        const tableInstance = table.instance() as any;
+        const updateStickyRow = jest.spyOn(tableInstance, "updateStickyRow");
```
Comment (martinmosko) [inline src/components/core/tests/PivotTable.spec.tsx]: this is `updateStickyRow, updateStickyHeadersPosition` mocked instances is shared between tests. I would say is not safe approach ...

Comment (yanlipnican) [reply src/components/core/tests/PivotTable.spec.tsx]: Yeah i should call mockRestore at end of test right?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by martinmosko** [comment]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3286/ : FAILURE in 8m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4068/ : SUCCESS in 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3287/ : SUCCESS in 22m 27s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4069/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1761/ : FAILURE in 10m 41s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3292/ : SUCCESS in 10m 47s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/899/ : SUCCESS in 1m 06s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/934/ : SUCCESS in 32m 13s

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1763/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1109/ : SUCCESS in 22m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/4081/ : SUCCESS in 55s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1765/ : UNSTABLE in 42m 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3296/ : SUCCESS in 8m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/900/ : SUCCESS in 29s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/936/ : SUCCESS in 18m 33s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1766/ : SUCCESS in 42m 07s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/3297/ : SUCCESS in 5m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/901/ : SUCCESS in 1m 09s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/938/ : SUCCESS in 22m 17s

**Pull request merged**